src/inference/include/openvino/runtime/intel_gpu/properties.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header for advanced hardware related properties for GPU plugin
+ *        To use in set_property, compile_model, import_model, get_property methods
+ *
+ * @file openvino/runtime/intel_gpu/properties.hpp
+ */
+#pragma once
+
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+/**
+ * @brief Read-only property which defines size of memory in bytes available for the device. For iGPU it returns host
+ * memory size, for dGPU - dedicated gpu memory size
+ */
+static constexpr Property<uint64_t, PropertyMutability::RO> device_total_mem_size{"GPU_DEVICE_TOTAL_MEM_SIZE"};
+
+/**
+ * @brief Read-only property to get microarchitecture identifier in major.minor.revision format
+ */
+static constexpr Property<std::string, PropertyMutability::RO> uarch_version{"GPU_UARCH_VERSION"};
+
+/**
+ * @brief Read-only property to get count of execution units for current GPU
+ */
+static constexpr Property<uint32_t, PropertyMutability::RO> execution_units_count{"GPU_EXECUTION_UNITS_COUNT"};
+
+/**
+ * @brief Read-only property to get statistics of GPU memory allocated by engine for each allocation type
+ * It contains information about current memory usage
+ */
+static constexpr Property<std::map<std::string, uint64_t>, PropertyMutability::RO> memory_statistics{
+    "GPU_MEMORY_STATISTICS"};
+
+/**
+ * @brief Read-only property to get maximum batch size which does not cause performance degradation due to memory swap
+ * impact.
+ */
+static constexpr Property<uint32_t, PropertyMutability::RO> max_batch_size{"GPU_MAX_BATCH_SIZE"};
+
```

Comment:
@myshevts @vladimir-paramuzov should not it be common OV property?

Reply:
agree. Let's move it to the common

Reply:
OPTIMAL_BATCH_SIZE is already common